hare/preroundtracker.go

```diff
@@ -49,21 +49,21 @@ func (pre *preRoundTracker) OnPreRound(ctx context.Context, msg *Msg) {
 	nodeID := types.BytesToNodeID(msg.PubKey.Bytes())
 
 	// check for winning VRF
-	sha := hash.Sum(msg.Eligibility.Proof)
-	shaUint32 := binary.LittleEndian.Uint32(sha[:4])
+	vrfHash := hash.Sum(msg.Eligibility.Proof)
+	vrfHashVal := binary.LittleEndian.Uint32(vrfHash[:4])
 	logger.With().Debug("received preround message",
 		nodeID,
 		log.String("sender_id", nodeID.ShortString()),
 		log.Int("num_values", len(msg.InnerMsg.Values)),
-		log.String("vrf_value", fmt.Sprintf("%x", shaUint32)))
+		log.Uint32("vrf_value", vrfHashVal))
 	// TODO: make sure we don't need a mutex here
```

Comment:
That's a serious TODO. I created an issue to track it. Would you mind sticking a link in the comment while you are at it? https://github.com/spacemeshos/go-spacemesh/issues/4007

Reply:
we dont need a mutex here. if you see algorithm.go, you will see messages are handled via a single channel

Reply:
removed this comment instead and commented on issue #4007